### PR TITLE
Works on a bunch of deploy jobs.

### PR DESCRIPTION
**Pull request by Cwagne17**
Partially completes [Story 8276](https://app.shortcut.com/clarkcan/story/8276/implement-circleci-orb).

Edits the setup-docker command.
Adds publish-docker functionality

### CODE REVIEW

File: src/commands/setup-docker.yml [hunk 1]
```diff
@@ -3,20 +3,20 @@ description: >
...
   docker-password:
-    type: env_var_name
     default: DOCKER_PASSWORD
-    description: >
-      Password for the docker account
+    description: |
+      Name of environment variable storing your Docker password
+    type: env_var_name
+  docker-username:
+    default: DOCKER_LOGIN
```
Comment (StarChaser44) [inline src/commands/setup-docker.yml:12]: Can we change this to DOCKER_USERNAME

File: src/jobs/deploy-ecs.yml [hunk 2]
```diff
@@ -1,9 +1,60 @@
...
+      Pick the language that the application is written in.
+      This value determines what type of executor should be used.
+  version:
+    type: string
+    default: 'latest'
+    description: >
+      Pick the version of the application to deploy to Dockerhub and ECS.
+  
+
 executor: node/default
```
Comment (StarChaser44) [inline src/jobs/deploy-ecs.yml:19]: Is this supposed to be dynamically set?

File: src/jobs/publish-docker.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+parameters:
+  docker-password:
+    type: env_var_name
+    default: DOCKER_PASSWORD
+    description: |
+      Name of environment variable storing your Docker password
+  docker-username:
+    type: env_var_name
+    default: DOCKER_LOGIN
```
Comment (StarChaser44) [inline src/jobs/publish-docker.yml:12]: Can this be DOCKER_USERNAME?

**Review by StarChaser44 — COMMENTED** [review]
I just had a few questions

**Review by EVogel1999 — COMMENTED** [review]
Just Tyree's comments

**Review by pzalep1 — COMMENTED** [review]
Just Tyree's questions

**Review by JosueMolinaMorales — DISMISSED** [review]
looks good

**Review by mfranz1 — DISMISSED** [review]

File: src/jobs/deploy-ecs.yml (hunk 2 above)
Comment (Cwagne17) [reply src/jobs/deploy-ecs.yml:54]: if it was statically set then every application would be deployed as the same version everytime.

**Review by EVogel1999 — APPROVED** [review]

File: src/commands/setup-aws.yml [hunk 4]
```diff
@@ -3,23 +3,42 @@ description: >
...
+  - run:
+      name: Verify Parameters
+      command: >
+        if [[ -z "$aws-secret-access-key" ]]; then
+          echo '"aws-secret-access-key" not set'
+          exit 1
+        fi
 
+        if [[ -z "$aws-access-key-id" ]]; then
+          echo '"aws-secret-access-key" not set'
```
Comment (mfranz1) [inline src/commands/setup-aws.yml:38]:
```suggestion
          echo '"aws-access-key-id" not set'
```

**Review by mfranz1 — COMMENTED** [review]
Just some syntax changes

**Review by StarChaser44 — APPROVED** [review]

**Review by JosueMolinaMorales — APPROVED** [review]

**Review by pzalep1 — DISMISSED** [review]

File: src/jobs/deploy-ecs.yml [hunk 5]
```diff
@@ -1,9 +1,119 @@
...
+      value, i.e. MACHINE_AWS_ECS_AccessKey.
+    type: env_var_name
+  aws-secret-access-key:
+    default: MACHINE_AWS_ECS_SecretKey
+    description: |
+      AWS secret key for IAM role. Set this to the name of
+      the environment variable you will use to hold this
+      value, i.e. $MACHINE_AWS_ECS_SecretKey.
+    type: env_var_name
+  mentions:
```
Comment (AStoerkel1) [inline src/jobs/deploy-ecs.yml:48]:
```suggestion
  slack-mentions:
```

**Review by EVogel1999 — APPROVED** [review]

File: src/jobs/deploy-ecs.yml [hunk 6]
```diff
@@ -1,9 +1,119 @@
 description: >
   This job deploys an application to aws elastic container service.
 
+parameters:
+  container-name:
+    type: string
```
Comment (JosueMolinaMorales) [inline src/jobs/deploy-ecs.yml:6]:
```suggestion
parameters:
  # Required Parameters
  container-name:
    type: string
```

File: src/jobs/deploy-ecs.yml [hunk 7]
```diff
@@ -1,9 +1,119 @@
...
+  cluster-name:
+    description: The short name or full ARN of the cluster that hosts the service.
+    type: string
+  service-name:
+    description: |
+      The name of the service to update. If undefined, we assume `family` is the
+      name of both the service and task definition.
+    type: string
+  isService:
+    type: boolean
```
Comment (JosueMolinaMorales) [inline src/jobs/deploy-ecs.yml:30]:
```suggestion
      name of both the service and task definition.
    type: string
  # Not Required Parameters
  isService:
    type: boolean
```

File: src/jobs/publish-docker.yml [hunk 8]
```diff
@@ -0,0 +1,76 @@
+description: >
+  This job publishes a new version of the application to DockerHub and versions the Github image.
+
+parameters:
+  docker-password:
```
Comment (JosueMolinaMorales) [inline src/jobs/publish-docker.yml:5]: Put Required Parameters at the top

**Review by JosueMolinaMorales — COMMENTED** [review]
just a couple of things

**Review by StarChaser44 — APPROVED** [review]

**Review by pzalep1 — APPROVED** [review]

**Pull request merged**